src/charm.py

```diff
@@ -134,10 +152,70 @@ def __init__(self, *args):
         )
         self.framework.observe(self.on.install, self._configure)
         self.framework.observe(self.on.collect_unit_status, self._on_collect_status)
+        self.framework.observe(self.on.collect_unit_status, self._collect_vault_app_status)
```

Comment:
I might have misunderstood what you meant in [this comment](https://github.com/canonical/vault-operator/pull/62#discussion_r1516350933).

I don't think collect_app_status is the correct place for it, since the app is running on each unit. We're talking about the vault app here, and not the Juju app, so the word is a little overloaded.

Anyways, if you have a better suggestion between these two approaches, I'm interested to hear, but just keeping this in `_on_collect_status` was putting us over the complexity limit which is why I originally broke it out there.

Reply:
ops offers two events that can be handled for status collection (`collect_unit_status` and `collect_app_status`) and using the word `app` here initially made me think you were trying to leverage the `app status`. If that's not the case , we should indeed avoid the word `app` here.

In any case here we are still observing `self.on.collect_unit_status` twice here which I don't understand. 


On the note of complexity, while I appreciate the warning and think it's often something we should care about, I think that some usecases are more readable as 1 function. For example this model scales quite well for status management:
```
if not x:
   event.add_status(BlockedStatus(xxx)):
   return
if not y:
   event.add_status(WaitingStatus(xxx)):
   return
...
event.add_status(ActiveStatus(xxx)):
```

Reply:
ok, I'll disable the warning

Reply:
Done in https://github.com/canonical/vault-operator/pull/62/commits/77183cd3714b0ccfa478b6d9edf664a3b2c3b8bf